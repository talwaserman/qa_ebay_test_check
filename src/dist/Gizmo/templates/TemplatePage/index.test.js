"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

var _context = require("./context");

var _index = _interopRequireDefault(require("./index"));

// test component rendering with different props
it('Gizmo shared - TemplatePage', function () {
  var tree = _reactTestRenderer.default.create( /*#__PURE__*/_react.default.createElement(_index.default, null)).toJSON();

  expect(tree).toMatchSnapshot();
}); // test reducer actions

it('Gizmo shared - TemplatePage - action add with valid value', function () {
  var todoList = ['buy dinner', 'run 5 km', 'finish work'];

  var updatedList = _context.actions.add(todoList, 'new value');

  var test = updatedList.length === 4 && updatedList.includes('new value');
  expect(test).toEqual(true);
});
it('Gizmo shared - TemplatePage - action add with invalid value', function () {
  var todoList = ['buy dinner', 'run 5 km', 'finish work'];

  var updatedList = _context.actions.add(todoList, '');

  var test = updatedList.length === 3 && !updatedList.includes('');
  expect(test).toEqual(true);
});