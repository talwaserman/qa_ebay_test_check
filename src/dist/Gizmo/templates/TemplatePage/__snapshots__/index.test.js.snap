// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gizmo shared - TemplatePage 1`] = `
<div
  className="template-page-wrapper"
>
  <h2>
    ** This is a template for page component **
  </h2>
  <div
    className="child-b-wrapper"
  >
    <input
      onChange={[Function]}
      value=""
    />
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <div
    className="child-a-wrapper"
  >
    <strong>
      List of todos:
    </strong>
    <div>
      buy dinner
    </div>
    <div>
      run 5 km
    </div>
    <div>
      finish work
    </div>
  </div>
  <div
    className="child-c-wrapper"
  >
    child c - not connected to context
  </div>
</div>
`;
